src/pykiso/lib/connectors/cc_grpc_client.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2023-2023 Accenture GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+Communication Channel Via GRPC
+********************************
+
+:module: cc_grpc_client
+
+:synopsis: Grpc communication channel
+
+.. currentmodule:: cc_grpc_client
+
+list of todo items:
+* Extend by supporting a path instead of files so that many proto files can be used
+* Extend with security capabilities
+
+"""
+
+import importlib.util
+import inspect
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+import grpc
+
+from pykiso import connector
+
+log = logging.getLogger(__name__)
+
+
+class GrpcClient(connector.CChannel):
+    """Grpc client implementation"""
+
+    def __init__(
+        self,
+        dest_ip: str,
+        dest_port: int,
+        generated_protobuf_data_types_file: str,
+        generated_protobuf_grpc_file: str,
+        default_service_name: str,
+        default_rpc_name: str,
+        default_message_name: str,
+        default_param: dict,
+        **kwargs,
+    ):
+        """Instantiate a grpc client
+
+        :param dest_ip (str): grpc server ip address
+        :param dest_port (int): grpc server port
+        :param generated_protobuf_data_types_file (str): path to the generated protobuf data types file
+        :param generated_protobuf_grpc_file (str): path to the generated protobuf grpc file
+        :param default_service_name (str): default service name to use
+        :param default_rpc_name (str): default rpc name to use
+        :param default_message_name (str): default message name to use
+        :param default_param (dict): default parameters to use in dictionary format
+        """
+        # Initialize the super class
+        super().__init__(**kwargs)
+        # Initialize the grpc client
+        self.dest_ip = dest_ip
+        self.dest_port = dest_port
+        # Import the generated datamodel protobuf file
+        spec = importlib.util.spec_from_file_location(
+            Path(generated_protobuf_data_types_file).stem,
+            Path(generated_protobuf_data_types_file),
+        )
+        self.grpc_datamodel = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(self.grpc_datamodel)
+        # Import the generated grpc protobuf file
+        # Import the module
+        spec = importlib.util.spec_from_file_location(
+            Path(generated_protobuf_grpc_file).stem, Path(generated_protobuf_grpc_file)
+        )
+        self.grpc_stub_module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(self.grpc_stub_module)
+        # Extract the stubs
+        module_attributes = inspect.getmembers(self.grpc_stub_module, inspect.isclass)
```

Comment:
what are you doing with inspect ? why do we need thos stubs ?

Reply:
https://grpc.io/docs/what-is-grpc/core-concepts/ -> search stub